apps/rush-lib/src/logic/operations/OperationStatus.ts

```diff
@@ -2,15 +2,40 @@
 // See LICENSE in the project root for license information.
 
 /**
- * Enumeration defining potential states of an operation: not started, executing, or completed
+ * Enumeration defining potential states of an operation
+ * @beta
  */
 export enum OperationStatus {
+  /**
+   * The Operation is on the queue, ready to execute (but may be waiting for dependencies)
+   */
   Ready = 'READY',
+  /**
+   * The Operation is currently executing
+   */
   Executing = 'EXECUTING',
+  /**
+   * The Operation completed successfully and did not write to standard output
+   */
   Success = 'SUCCESS',
+  /**
+   * The Operation completed successfully, but wrote to standard output
+   */
   SuccessWithWarning = 'SUCCESS WITH WARNINGS',
+  /**
+   * The Operation was skipped via the legacy incremental build logic
+   */
   Skipped = 'SKIPPED',
+  /**
+   * The Operation had its outputs restored from the build cache
+   */
   FromCache = 'FROM CACHE',
```

Comment:
This one still seems to stick out to me as being not in-keeping with what the status' are for. Kinda in the same vein as splitting the caching stuff out of the OperationExecutionManager. Not blocking, since it's maintaining existing functionality, but maybe something to consider before making this API public

Reply:
Will need some thought, definitely.